src/cpp/src/lm_encoding.cpp

```diff
@@ -130,7 +131,7 @@ std::pair<EncodedResults, std::optional<int64_t>> get_lm_encoded_results(
         beam_offets.insert({sequence_groups.at(i)->get_request_id(), i});
 
     SamplerOutput sampler_output = sampler.sample(sequence_groups, logits);
-    stream_generated_tokens();
```

Comment:
probably, if we start to stream tokens here in a dedicated thread, it can be faster as streaming will be overlapped with:
- embedding model
- llm model for next token
- sampling for next token

while currently streaming is overlapped with LLM only

we could re-use `SynchronizedQueue` from GenAI source where streaming callback will push to queue, streaming dedicated thread will read from this queue using `pull` method